### PR TITLE
Rewrite Cbc wrapper with Clang.

### LINKED ISSUE #122: Cbc backend returns primal status feasible for infeasible problem
Sorry if this is the wrong place for this.

MWE with a constraint that 10 binary variables shall add up to 11:

```julia
julia> using JuMP, Cbc

julia> mmm = Model(with_optimizer(Cbc.Optimizer));

julia> @variable(mmm, x[1:10], Bin);

julia> @objective(mmm, Max, sum(x))
x[1] + x[2] + x[3] + x[4] + x[5] + x[6] + x[7] + x[8] + x[9] + x[10]

julia> @constraint(mmm, infeasible, sum(x) == 11)
infeasible : x[1] + x[2] + x[3] + x[4] + x[5] + x[6] + x[7] + x[8] + x[9] + x[10] == 11.0

julia> JuMP.optimize!(mmm)
Welcome to the CBC MILP Solver
Version: 2.9.9
Build Date: Dec 31 2018

command line - Cbc_C_Interface -solve -quit (default strategy 1)
Problem is infeasible - 0.00 seconds
Total time (CPU seconds):       0.01   (Wallclock seconds):       0.01


julia> JuMP.termination_status(mmm)
INFEASIBLE::TerminationStatusCode = 2

julia> JuMP.primal_status(mmm)
FEASIBLE_POINT::ResultStatusCode = 1
````

Also sorry if this is intended/unavoidable behaviour or if I'm just misinformed (not very experienced with optimization).

For the real problem I am interested in a feasible solution even though it is not optimal so I ran into the issue because I only checked the primal status. 

I guess it is enough to look at the termination status to determine that the solver considered the problem infeasible and only look at the primal status if not infeasible.

### LINKED ISSUE #124: Support SolveTime
See https://www.juliaopt.org/MathOptInterface.jl/v0.9.1/apireference/#MathOptInterface.SolveTime

### PR DESCRIPTION
**Pull request by odow**
This PR also includes a number of style improvements to the MOI wrapper.

Closes #122 
Closes #124

### CODE REVIEW

File: src/Cbc.jl [hunk 1]
```diff
@@ -1,25 +1,43 @@
...
-# This 'using' is required to suppress a warning about Clp not having Libdl in
-# its dependencies (Libdl is used by BinaryProvider), e.g.:
-# bicycle1885/CodecZlib.jl#26.
-using Libdl
-
 if haskey(ENV,"JULIA_CBC_LIBRARY_PATH") || VERSION < v"1.3"
-    if isfile(joinpath(@__DIR__, "..", "deps", "deps.jl"))
-        include(joinpath(@__DIR__, "..", "deps", "deps.jl"))
+    if isfile(joinpath(dirname(@__FILE__),"..","deps","deps.jl"))
+        include("../deps/deps.jl")
```
Comment (blegat) [inline src/Cbc.jl:5]: Why don't we use the same path as the above line?

**Comment by blegat** [comment]
The `moi` folder is inconsistent with https://www.juliaopt.org/MathOptInterface.jl/stable/apimanual/#Naming-1
According to it, since there is only one file, it should just be `src/MOI_wrapper.jl`.

**Comment by odow** [comment]
I fixed the folder to `MOI`. I'd prefer to keep it in a folder, even though at the moment it is only one file. We may add extra features going forward.

**Comment by odow** [comment]
> Can't see the diff of MOI_wrapper.jl, could you do the file renaming in a separate PR?

Done.

**Comment by blegat** [comment]
It might help to add in the title of first post of this PR that this also drops MPB.

**Pull request merged**